### PR TITLE
Handle open, deploy & run of Zeebe diagrams on usage statistics

### LINKED ISSUE #2115: Make UsageStatistics aware of the Zeebe tab
__What should we do?__

* [ ] Correctly attach metadata to diagram.opened events
* [ ] Collect deployment data for Zeebe diagrams

__Why should we do it?__

We want to integrate Zeebe into Camunda Modeler.

Child of https://github.com/bpmn-io/internal-docs/issues/216

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #2115 

For example events refer to https://github.com/camunda/camunda-modeler/pull/2123#issuecomment-788034342.

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
**Example events**

* a Camunda BPMN diagram was opened

```js
{
  "event": "diagramOpened",
  "diagramType": "bpmn",
  "elementTemplates": [ ... ],
  "elementTemplateCount": 16,
  "diagramMetrics": {
    "processVariablesCount": 0,
    "tasks": {
      "userTask": {
        "count": 0,
        "form": {
          "count": 0,
          "embedded": 0,
          "external": 0,
          "generated": 0,
          "other": 0
        }
      }
    }
  },
  "engineProfile": {
    "executionPlatform": "Camunda Platform",
    "executionPlatformVersion": "7.15.0"
  }
}
```

* a Camunda BPMN diagram was deployed

```js
{
  "event": "deployment",
  "diagramType": "bpmn",
  "deployment": {
    "outcome": "success",
    "context": "deploymentTool"
  },
  "diagramMetrics": {
    "processVariablesCount": 0,
    "tasks": {
      "userTask": {
        "count": 0,
        "form": {
          "count": 0,
          "embedded": 0,
          "external": 0,
          "generated": 0,
          "other": 0
        }
      }
    }
  },
  "engineProfile": {
    "executionPlatform": "Camunda Platform",
    "executionPlatformVersion": "7.15.0"
  }
}
```

* a Zeebe BPMN diagram was opened

```js
{
  "event": "diagramOpened",
  "diagramType": "bpmn",
  "diagramMetrics": {
    "processVariablesCount": 0,
    "tasks": {
      "userTask": {
        "count": 0,
        "form": {
          "count": 0,
          "camundaForms": 0,
          "other": 0
        }
      }
    }
  },
  "engineProfile": {
    "executionPlatform": "Camunda Cloud",
    "executionPlatformVersion": "1.0.0"
  }
}
```

* a Zeebe BPMN diagram was deployed

```js
{
  "event": "deployment",
  "diagramType": "bpmn",
  "deployment": {
    "outcome": "failure",
    "context": "deploymentTool",
    "error": "INVALID_ARGUMENT",
    "targetType": "camundaCloud"
  },
  "diagramMetrics": {
    "processVariablesCount": 0,
    "tasks": {
      "userTask": {
        "count": 4,
        "form": {
          "count": 4,
          "camundaForms": 4,
          "other": 0
        }
      }
    }
  },
  "engineProfile": {
    "executionPlatform": "Camunda Cloud",
    "executionPlatformVersion": "1.0.0"
  }
}
```

**Comment by barmac** [comment]
I've rebased this on top of `develop` and am reviewing this right now.

**Comment by barmac** [comment]
We don't count the variables for Zeebe diagrams, do we? It always sends `"processVariablesCount": 0` in my case.

**Comment by pinussilvestrus** [comment]
> We don't count the variables for Zeebe diagrams, do we? It always sends "processVariablesCount": 0 in my case.

That's true, we currently don't have the ability to extract process variables for Zeebe diagrams, that's why we set it to zero (cf. https://github.com/camunda/camunda-modeler/pull/2123/files#diff-023b55f2c41ce07c321f1744251ca781caeb028a7a360c5ea4724fb648144ea0R22). I considered setting it to null or simply not include the field

```js
{
  // (a)
  processVariablesCount: 0,

  // (b)
  processVariablesCount: null,

  // (c)
  <don't include it>
}
```

but decided to go with the _zero_ to avoid unknown field types on E.T, since it was always a number before. What do you think @volkergersabeck @barmac ?

**Comment by barmac** [comment]
If excluding this field from the payload could cause problems, I'm OK with `0` as the value.

**Comment by pinussilvestrus** [comment]
Things to change before merging it:
* Remove `executionPlatformVersion` for now
* Remove `processVariablesCount` for zeebe diagrams for now

Both metrics doesn't add any value for now, since they are wrong / test data / not properly tracked.

**Comment by pinussilvestrus** [comment]
@barmac should be fine now

**Comment by barmac** [comment]
I noticed that we send empty engineProfile for Camunda diagrams. I believe we should have a default value there - correct me if I'm wrong.

**Comment by pinussilvestrus** [comment]
I wouldn't set default values on the usage statistics level, but rather ensure we have a default value set when creating new diagrams. Setting a default value on the event sending level would mean we assume every old diagram points to the Camunda Platform, which we maybe can't. I'd rather not sending anything instead of default values.

File: client/src/app/tabs/bpmn/diagram.bpmn [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" id="Definitions_{{ ID }}" targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="3.0.0-dev">
+<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" id="Definitions_{{ ID }}" targetNamespace="http://bpmn.io/schema/bpmn" xmlns:modeler="http://camunda.org/schema/modeler/1.0" modeler:executionPlatform="Camunda Platform" modeler:executionPlatformVersion="7.14.0">
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/diagram.bpmn:2]: Btw. this also fixes https://github.com/camunda/camunda-modeler/issues/2030 

😉

Comment (barmac) [reply client/src/app/tabs/bpmn/diagram.bpmn:2]: True, but is the exporter set correctly?

Comment (pinussilvestrus) [reply client/src/app/tabs/bpmn/diagram.bpmn:2]: I guess only on first save / export.

Comment (barmac) [reply client/src/app/tabs/bpmn/diagram.bpmn:2]: OK, so we need to first change the diagram in order to set the value. I guess it's OK then.

**Comment by pinussilvestrus** [comment]
@barmac I added another commit (cc7493ac8f4d9cfb5695ede95d2602af30715da5) on top to ensure we always send an execution platform, also for legacy diagrams. I think this in line now with our discussions with @volkergersabeck.

**Review by barmac — APPROVED** [review]

**Pull request merged**